app/database/database_service.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require_relative './cache_service'
+
+class DatabaseService
+  def initialize(db_file_path = nil)
+    records = nil
+    @db_file_path = db_file_path || File.dirname(__FILE__) + '/db.txt'
+    File.open(@db_file_path, 'r') do |f|
+      records = f.map { |line| deserialize(line) }
```

Comment:
shorthand this like `records = f.map(&:deserialize)`, that just applies the `deserialize` method to each element in the array and returns the result.